internal/resources/aws/cloudfront_distribution.go

```diff
@@ -0,0 +1,659 @@
+			monthlyHTTPRequests:             r.MonthlyHTTPRequests.AsiaPacific,
+			monthlyHTTPSRequests:            r.MonthlyHTTPSRequests.AsiaPacific,
+			monthlyDataTransferToInternetGB: r.MonthlyDataTransferToInternetGB.AsiaPacific,
+			monthlyDataTransferToOriginGB:   r.MonthlyDataTransferToOriginGB.AsiaPacific,
+		},
+		{
+			awsGroupedName:                  "India",
+			priceRegion:                     "India",
+			monthlyHTTPRequests:             r.MonthlyHTTPRequests.India,
+			monthlyHTTPSRequests:            r.MonthlyHTTPSRequests.India,
+			monthlyDataTransferToInternetGB: r.MonthlyDataTransferToInternetGB.India,
+			monthlyDataTransferToOriginGB:   r.MonthlyDataTransferToOriginGB.India,
+		},
+	}
+
+	subresources := []*schema.Resource{}
+
+	for _, data := range regionsData {
+		if !data.HasUsage() {
+			continue
+		}
+
+		subresources = append(subresources, r.buildRegionSubresource(data))
+	}
+
+	if len(subresources) == 0 {
+		subresources = append(subresources, r.buildRegionSubresource(regionsData[0]))
+	}
+
+	return subresources
+}
+
+func (r *CloudfrontDistribution) buildRegionSubresource(regionData *cloudfrontDistributionRegionData) *schema.Resource {
+	resource := &schema.Resource{
+		Name: regionData.awsGroupedName,
+	}
+
+	components := []*schema.CostComponent{}
+	components = append(components, r.dataOutToInternetCostComponents(regionData)...)
+	components = append(components, r.dataOutToOriginCostComponents(regionData)...)
+	components = append(components, r.httpRequestsCostComponents(regionData)...)
+	components = append(components, r.httpsRequestsCostComponents(regionData)...)
+
+	resource.CostComponents = components
+
+	return resource
+}
+
+func (r *CloudfrontDistribution) dataOutToInternetCostComponents(regionData *cloudfrontDistributionRegionData) []*schema.CostComponent {
+	tierStarts := []int{0, 10240, 51200, 153600, 512000, 1048576, 5242880}
+	tierLimits := []int{10240, 40960, 102400, 358400, 536576, 4194304}
+	tierNames := []string{"first 10TB", "next 40TB", "next 100TB", "next 350TB", "next 524TB", "next 4PB", "over 5PB"}
+
+	fromLocation := regionData.priceRegion
+
+	var quantity *decimal.Decimal
+	if regionData.monthlyDataTransferToInternetGB != nil {
+		quantity = decimalPtr(decimal.NewFromFloat(*regionData.monthlyDataTransferToInternetGB))
+	}
+
+	if quantity == nil {
+		return []*schema.CostComponent{
+			r.buildDataOutCostComponent(tierNames[0], fromLocation, 0, nil),
+		}
+	}
+
+	tiers := usage.CalculateTierBuckets(*quantity, tierLimits)
+	var components []*schema.CostComponent
+	for i := range tiers {
+		if tiers[i].GreaterThan(decimal.Zero) {
+			components = append(
+				components,
+				r.buildDataOutCostComponent(tierNames[i], fromLocation, tierStarts[i], &tiers[i]),
+			)
+		}
+	}
+
+	return components
+}
+
+func (r *CloudfrontDistribution) buildDataOutCostComponent(usageName, fromLocation string, startUsage int, quantity *decimal.Decimal) *schema.CostComponent {
```

Comment:
Not suggesting we change this, just wondering why you're using `build` prefix for this cost component method?

Reply:
Mostly to distinguish `dataOutToInternetCostComponents` function name and `buildDataOutCostComponent` that is used in the first one. It should not be used as standalone cost components. Otherwise `dataOutToInternetCostComponents` and `dataOutCostComponent` would look pretty similar and it would be hard to understand which one to include in resource's cost components.

Reply:
👍